GVRf/Framework/framework/src/main/jni/engine/renderer/batch.cpp

```diff
@@ -87,47 +128,51 @@ bool Batch::add(RenderData *render_data) {
             renderdata_ = new RenderData(*render_data);
             renderdata_->set_batching(true);
        }
-
     }
 
     render_data_set_.insert(render_data); // store all the renderdata which are in batch
     render_mesh->setMeshModified(false); // mark mesh clean
 
-    const std::vector<glm::vec3>& vertices = render_mesh->vertices();
-    const std::vector<glm::vec3>& normals = render_mesh->normals();
-    const std::vector<glm::vec2>& tex_coords = render_mesh->tex_coords();
-    int size = 0;
+    updateMesh(render_mesh);
 
-    size = vertices.size();
-    for (int i=0; i < size; i++) {
-        vertices_.push_back(vertices[i]);
-        tex_coords_.push_back(tex_coords[i]);
-        matrix_indices_.push_back(draw_count_);
-    }
-
-    size = normals.size();
-    for (int i=0; i < size; i++) {
-        normals_.push_back(normals[i]);
-    }
-
-    size = indices.size();
-    index_count_+=size;
-    for (int i = 0; i < size; i++) {
-        unsigned short index = indices[i];
-        index += index_offset_;
-        indices_.push_back(index);
+    return true;
+}
+void Batch::clearData(){
+    vertex_count_ = 0;
+    index_count_ = 0;
+    index_offset_ = 0;
+    draw_count_=0;
+    matrix_index_map_.clear();
+    matrix_indices_.clear();
+    matrices_.clear();
+    tex_coords_.clear();
+    vertices_.clear();
+    normals_.clear();
+}
+bool Batch::isRenderDataDisabled(){
+     bool update_vbo = false;
+     for(auto it= render_data_set_.begin();it!=render_data_set_.end();){
+        if(!(*it)->enabled() || !(*it)->owner_object()->enabled()){
+            (*it)->set_batching(false);
```

Comment:
Not related. Actually first we check whether batch is null for a renderdata if it is then we try to add into available or new batch. So in add() we return if we have disabled batching.